lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+          loaded_spec_files = RSpec.configuration.loaded_spec_files
+
+          if loaded_spec_files.include?(new_definition) && old_definition == new_definition
```

Comment:
What we're checking for here isn't that it's a duplicate definition, actually, but a single definition of a shared example group that's loaded twice, which was saying that there was a duplicate definition when there wasn't _really_ a duplication. For example, here's the warning I was seeing in my tests:

```
WARNING: Shared example group 'a flagger_or_tagger' has been previously defined
at:
  /Users/devoncestes/esh/IRT/spec/services/shared_examples/flagger_or_tagger_spec.rb:10
...and you are now defining it at:
  /Users/devoncestes/esh/IRT/spec/services/shared_examples/flagger_or_tagger_spec.rb:10
The new definition will overwrite the original one.
```

I didn't have a duplicate definition there, but because of the naming of the file combined with the explicit require that we were doing, it was evaluating that shared example group definition twice even though it was only defined once.